engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/activity/UserTaskTest.java

```diff
@@ -299,6 +299,31 @@ public void shouldCreateJobWithEvaluatedAssigneeExpressionHeader() {
         .containsEntry(Protocol.USER_TASK_ASSIGNEE_HEADER_NAME, "alice");
   }
 
+  @Test
+  public void shouldCreateJobAndIgnoreEmptyEvaluatedAssigneeExpressionHeader() {
+    // given
+    ENGINE.deployment().withXmlResource(process(t -> t.zeebeAssigneeExpression("user"))).deploy();
+
+    // when
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withVariables(Map.of("user", ""))
+            .create();
+
+    // then
+    final Record<JobRecordValue> job =
+        RecordingExporter.jobRecords(JobIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    final Map<String, String> customHeaders = job.getValue().getCustomHeaders();
+    assertThat(customHeaders)
+        .hasSize(0)
+        .doesNotContainEntry(Protocol.USER_TASK_ASSIGNEE_HEADER_NAME, "alice");
```

Comment:
Sorry for the drive-by review but this looks like a copy-paste mistake. It should probably be something like `doesNotContainKey(Protocol.USER_TASK_ASSIGNEE_HEADER_NAME)`

Reply:
You are correct 🙈 I'll modify it. It still tests the same thing because of the `hasSize(0)` but this is definitely more clear.